Establish project-wide contribution guidelines

dhimmel opened the pull request:
`CONTRIBUTING.md` provides high-level guidelines for contributing to Project Cognoma repositories.

Specific respositories, such as [`django-cognoma`](https://github.com/cognoma/django-cognoma) may want to add their own `CONTRIBUTING.md` files that link to these guidelines and then contain repo-specific guidelines.

I'm looking for feedback before merging these guidelines.

dhimmel committed:
Add contribution info

awm33 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,30 @@
+
+## Pull Requests
+
+Project Cognoma operates on a pull request model. Substantial changes should be submitted via [pull request](https://help.github.com/articles/using-pull-requests/ "GitHub · Using pull requests"). We also encourage you to comment on and review open pull requests.
```
You may want to state "all" or "any" changes, not just substantial ones. Hot fixes or emergencies may need to bypass PRs, but that should be rare.

cgreene commented:
I agree with @awm33. As this is the project repository, I don't think we'll have hot fixes here. It'd be great for the component repositories to define their own policies in those cases.

cgreene committed:
suggested changes based on @awm33

also makes it clear that component projects can customize the model to their needs.

cgreene committed:
cleanup

dhimmel committed:
Incorporate all changes via PR policy

dhimmel commented:
Okay @cgreene I added a few changes on top of yours to the "Commit Access" section. These changes reflect the new policy of mandatory PRs for `cognoma/cognoma`.
